### DISCUSSION TITLE
Unable to use certain classes from .NET

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/fsharp#16870] [Q&A] [answered]
Microsoft.AspNetCore.Components.NavigationOptions is available for F# through nuget package, yet attempting to construct the object in fsharp is impossible due to it looking like this
```F#
///<summary>
///            Additional options for navigating to another URI.
///            </summary>
[<Struct>]
type NavigationOptions =
  
  val ``<ForceLoad>k__BackingField`` : bool
  
  val ``<HistoryEntryState>k__BackingField`` : string
  
  val ``<ReplaceHistoryEntry>k__BackingField`` : bool
  
  ///<summary>
  ///            If true, bypasses client-side routing and forces the browser to load the new page from the server, whether or not the URI would normally be handled by the client-side router.
  ///            </summary>
  member ForceLoad: bool
  
  ///<summary>
  ///            Gets or sets the state to append to the history entry.
  ///            </summary>
  member HistoryEntryState: string
  
  ///<summary>
  ///            If true, replaces the currently entry in the history stack.
  ///            If false, appends the new entry to the history stack.
  ///            </summary>
  member ReplaceHistoryEntry: bool
```
Making interop to blazor from F# really difficult all the sudden

**Comment by vzarytovskii** [comment, marked as answer]
You can, unless I'm missing something
<img width="1276" alt="image" src="https://github.com/dotnet/fsharp/assets/1260985/375d4fb8-49e3-4374-9acb-b7c018fa982b">

**Reply by Xyncgas** [reply]
You are correct. I used Visual Studio's intellisense to learn what I can do but I didn't know I can do it this way, it's my problem

I attempted to :
```F#
    let x = NavigationOptions()
    x.ForceLoad <- true
````

The initial impression I received for seeing an F# struct without mutable fields/properties and having no constructor shocked me
